source/dem/dem.cc

```diff
@@ -199,21 +199,31 @@ DEMSolver<dim>::DEMSolver(DEMSolverParameters<dim> dem_parameters)
                                            simulation_control->get_time_step());
 
   for (unsigned int i_solid = 0;
-       i_solid < parameters.solid_objects->number_solids;
+       i_solid < parameters.solid_objects->number_solid_surfaces;
        ++i_solid)
     {
-      solids.push_back(std::make_shared<SerialSolid<dim - 1, dim>>(
-        this->parameters.solid_objects->solids[i_solid], i_solid));
+      solid_surfaces.push_back(std::make_shared<SerialSolid<dim - 1, dim>>(
+        this->parameters.solid_objects->solid_surfaces[i_solid], i_solid));
+    }
+
+  for (unsigned int i_solid = 0;
+       i_solid < parameters.solid_objects->number_solid_volumes;
+       ++i_solid)
+    {
+      solid_volumes.push_back(std::make_shared<SerialSolid<dim, dim>>(
+        this->parameters.solid_objects->solid_volumes[i_solid], i_solid));
     }
 
   // Generate solid objects
-  floating_mesh_info.resize(solids.size());
+  solid_surfaces_mesh_info.resize(solid_surfaces.size());
+  solid_volumes_mesh_info.resize(solid_volumes.size());
 
   // Resize particle_floating_mesh_in_contact
-  if (solids.size() > 0)
+  if ((solid_surfaces.size() + solid_volumes.size()) > 0)
     {
-      has_floating_mesh = true;
-      contact_manager.particle_floating_mesh_in_contact.resize(solids.size());
+      has_solid_objects = true;
+      contact_manager.particle_floating_mesh_in_contact.resize(
+        solid_surfaces.size() + solid_volumes.size());
```

Comment:
I guess those are gonna be handles differently with the implementation of particle-solid volume contacts?

Reply:
yes probably. I didn't put a lot of time into the contact detection with  the particles, so I'm still not sure how this is gonna work.

Reply:
I think it is fine for now

Reply:
I think too